### PR TITLE
EOS-20665: Field Prepare - Storage configuration

### PR DESCRIPTION
**Pull request by 83bhp**
Signed-off-by: Pritam Bhavsar <pritam.bhavsar@seagate.com>

### CODE REVIEW

File: lr-cli/cortx_setup/commands/node/prepare/storage.py [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+
+        _node_id = local_minion_id()
+        _enc_num = "enclosure-" + ((_node_id).split('-'))[1]
+        _enc_id_on_node = Commons().fetch_enc_id(_node_id)
+        _enc_user = kwargs.get('user')
+        _enc_passwd = kwargs.get('password')
+
+ 
+        try:
+            validate_credentials(_node_id, _enc_num, _enc_user, _enc_passwd)
```
Comment (divya-vijay) [inline lr-cli/cortx_setup/commands/node/prepare/storage.py:125]: `self.validate_credentials`

File: lr-cli/cortx_setup/commands/node/prepare/storage.py [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+            'optional': True
+        },
+        'password': {
+            'type': str,
+            'default': None,
+            'optional': True
+        }
+    }
+
+    def validate_user_password(
```
Comment (divya-vijay) [inline lr-cli/cortx_setup/commands/node/prepare/storage.py:42]: I guess the method name is `validate_credentials`

File: lr-cli/cortx_setup/commands/node/prepare/storage.py [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+
+from cortx_setup.commands.command import Command
+from cortx_setup.commands.storage import Commons
+from cortx_setup.config import CONFSTORE_CLUSTER_FILE
+from cortx_setup.commands.common_utils import get_pillar_data
+from provisioner.commands import PillarSet
+from provisioner.salt import local_minion_id
+from provisioner.salt import function_run
+
+
```
Comment (divya-vijay) [inline lr-cli/cortx_setup/commands/node/prepare/storage.py:26]: Imports to be added:
```
from cortx.utils.conf_store import Conf
from cortx.utils.security.cipher import Cipher
```

File: lr-cli/cortx_setup/commands/node/prepare/storage.py [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+from cortx_setup.commands.command import Command
+from cortx_setup.commands.storage import Commons
```
Comment (divya-vijay) [inline lr-cli/cortx_setup/commands/node/prepare/storage.py:19]: file name in import 
`from cortx_setup.commands.storage.commons import Commons`

File: lr-cli/cortx_setup/commands/node/prepare/storage.py [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+                " different than the one that was used in factory."
+            )
+            self.logger.info(
+                "Updating the new enclosure id on the node"
+            )
+            self.logger.debug(
+                f"Updating the enclosure id {_enc_id_on_enc} in pillar"
+            )
+            PillarSet().run(
+                'storage/{_enc_num}/enclosure_id',
```
Comment (divya-vijay) [inline lr-cli/cortx_setup/commands/node/prepare/storage.py:166]: f-string.
`f'storage/{_enc_num}/enclosure_id'`

File: lr-cli/cortx_setup/commands/node/prepare/storage.py [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+            _user_check = False
+        else:
+            self.logger.debug(
+                f"Username provide [{user} matches with"
+                " the user name in configuration"
+            )
+
+        if passwd != _ctrl_secret:
+            self.logger.warning(
+                f"Password provided [{passwd} does not match"
```
Comment (divya-vijay) [inline lr-cli/cortx_setup/commands/node/prepare/storage.py:95]: All warning and debug msgs have array in between `[ ]` . can be removed.

File: lr-cli/cortx_setup/commands/__init__.py [hunk 7]
```diff
@@ -29,6 +29,7 @@
 from .node.prepare.firewall import NodePrepareFirewall
 from .node.prepare.network import NodePrepareNetwork
 from .node.prepare.time import NodePrepareTime
+from .node.prepare.time import NodePrepareStorage
```
Comment (divya-vijay) [inline lr-cli/cortx_setup/commands/__init__.py:32]: `from .node.prepare.storage import NodePrepareStorage`

**Review by divya-vijay — CHANGES_REQUESTED** [review]

File: lr-cli/cortx_setup/commands/__init__.py (hunk 7 above)
Comment (83bhp) [reply lr-cli/cortx_setup/commands/__init__.py]: done.

File: lr-cli/cortx_setup/commands/node/prepare/storage.py (hunk 6 above)
Comment (83bhp) [reply lr-cli/cortx_setup/commands/node/prepare/storage.py]: done

File: lr-cli/cortx_setup/commands/node/prepare/storage.py (hunk 5 above)
Comment (83bhp) [reply lr-cli/cortx_setup/commands/node/prepare/storage.py]: done

File: lr-cli/cortx_setup/commands/node/prepare/storage.py (hunk 4 above)
Comment (83bhp) [reply lr-cli/cortx_setup/commands/node/prepare/storage.py]: done

File: lr-cli/cortx_setup/commands/node/prepare/storage.py (hunk 3 above)
Comment (83bhp) [reply lr-cli/cortx_setup/commands/node/prepare/storage.py:28]: done

File: lr-cli/cortx_setup/commands/node/prepare/storage.py (hunk 2 above)
Comment (83bhp) [reply lr-cli/cortx_setup/commands/node/prepare/storage.py]: done

File: lr-cli/cortx_setup/commands/node/prepare/storage.py (hunk 1 above)
Comment (83bhp) [reply lr-cli/cortx_setup/commands/node/prepare/storage.py]: done

**Review by ypise — APPROVED** [review]

**Pull request merged**